lib/gRpcServer.js

```diff
@@ -40,26 +41,43 @@ class GRpcServer {
   }
 
   run() {
-    const server = new grpc.Server();
-
+    this.server = new grpc.Server();
     this.configureRouter();
 
     this.services = Definition.loadServices(this.packageName);
 
     Object.keys(this.services).forEach((resource) => {
       if (app.controllers[`${resource}Controller`]) {
-        const handlers = getMethodNames(app.controllers[`${resource}Controller`]).reduce((mappings, key) => {
+        const handlers = getMethodNames(
+          app.controllers[`${resource}Controller`]
+        ).reduce((mappings, key) => {
           mappings[key] = this.generateRequestHandler(resource, key);
           return mappings;
         }, {});
 
-        server.addService(this.services[resource].service, handlers);
+        this.server.addService(this.services[resource].service, handlers);
       }
     });
 
-    server.bind(`${this.ip}:${this.port}`, grpc.ServerCredentials.createInsecure());
+    this.server.bind(
+      `${this.ip}:${this.port}`,
+      grpc.ServerCredentials.createInsecure()
+    );
     log('sl-express', 'info', { message: `Starting grpc server on 0.0.0.0:${this.port}` });
-    return server.start();
+    return this.server.start();
+  }
+
+  stop() {
+    return new Promise((resolve, reject) => {
+      if (this.server) {
```

Comment:
maybe add a `setTimeout` and call `reject` for setting a max wait time for gracefully shutdown

Reply:
should check `this.server` outside otherwise it may never resolve